core/src/main/java/flatgraph/util/DiffTool.java

```diff
@@ -0,0 +1,142 @@
+package flatgraph.util;
+
+import flatgraph.*;
+import java.util.*;
+
+import scala.jdk.javaapi.CollectionConverters;
+
+public class DiffTool {
+
+  /** compare two graphs element by element
+   * identity of nodes is given by their id, i.e. if the graphs have the same nodes/edges/properties, but use different
+   * node ids, a lot of differences will be reported and the results will be useless
+   */
+  public static List<String> compare(Graph graph1, Graph graph2) {
+    final List<String> diff = new ArrayList<>();
+    if (graph1.nodeCount() != graph2.nodeCount()) {
+      diff.add(String.format("node count differs: graph1=%d, graph2=%d", graph1.nodeCount(), graph2.nodeCount()));
+    }
+    
+    SortedSet<Long> nodeIds = new TreeSet<>();
+    graph1.allNodes().foreach(node -> nodeIds.add(node.id()));
```

Comment:
This can be much simpler and faster by just directly accessing the graph's underlying node arrays.

But if this works then it's fine with me, we can improve it later.

Reply:
that's just a copy of the odb code, and given that it works I'd concur with 'improve later'